src/gf.c

```diff
@@ -749,6 +749,7 @@ static int check_ambiguous_visitor(jl_typemap_entry_t *oldentry, struct typemap_
     struct ambiguous_matches_env *closure = container_of(closure0, struct ambiguous_matches_env, match);
     if (oldentry == closure->newentry)
         return 0; // finished once it finds the method that was just inserted
+    // TODO: instead, maybe stop once we hit something newentry is definitely more specific than
```

Comment:
This was motivated by the following example:

```
f(::Int8, x)
f(::Integer, x)
f(x, ::Int)
```

Here we fail to detect that `Int8, Int` is ambiguous.

Yes, that sounds promising.